gitapp/lib/users_list/home_page_view.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'package:gitapp/users_list/searchbar_widget.dart';
+import 'package:gitapp/users_list/connection_presenter.dart';
+import 'package:gitapp/users_list/user_cards_widget.dart';
+import 'package:gitapp/users_list/users_presenter.dart';
+import 'no_connection_widget.dart';
+
+class HomePage extends StatefulWidget {
+  HomePage({Key? key, required this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  HomePageState createState() => HomePageState();
+}
+
+class HomePageState extends State<HomePage> with TickerProviderStateMixin {
+  final UsersPresenter users = UsersPresenter();
+  late Searchbar searchBar = new Searchbar(getUsers);
+  bool connectionStatus = true;
+  bool searching = false;
+
+  Future getUsers(String text) async {
+    connectionStatus = await Connection.instance.checkConnection();
+    setState(() {});
+    if (text != "" || !connectionStatus) {
+      setState(() {
+        searching = true;
+      });
+      await users.updateSearchedListFromApi(text);
+      setState(() {
+        searchBar.textController.clear();
+        searching = false;
+      });
+    } else
+      users.searchedList.clear();
+  }
+
+  Widget chooseWidget() {
```

Comment:
What does it mean `chooseWidget` ? Choose widget from what? The name of this method is not providing us the proper context and someone which doest not know the code won't understand this.

Reply:
Also this logic should be moved to presenter. This is logic connected with connectionStatus. Not in the view

Reply:
> Also this logic should be moved to presenter. This is logic connected with connectionStatus. Not in the view

Connection is now in presenter.

Reply:
> What does it mean `chooseWidget` ? Choose widget from what? The name of this method is not providing us the proper context and someone which doest not know the code won't understand this.

name changed to chooseWidgetBasedOnState